build/source/driver/multi_driver.f90

```diff
@@ -384,130 +433,139 @@ real(dp),parameter        :: safety = 0.01_dp               ! safety factor to e
-                 fluxStruct%hru(iHRU),    & ! data structure of model fluxes 
-                 err,message)               ! error control
+ ! calculate the fraction of runoff in future time steps
+ call fracFuture(bparStruct%gru(iGRU)%var,    &  ! vector of basin-average model parameters
+                 bvarStruct%gru(iGRU),        &  ! data structure of basin-average variables
+                 err,message)                    ! error control
  call handle_err(err,message)
 
- ! calculate "short-cut" variables such as volumetric heat capacity
- call v_shortcut(mparStruct%hru(iHRU)%var,& ! vector of model parameters
-                 diagStruct%hru(iHRU),    & ! data structure of model diagnostic variables
-                 err,message)               ! error control
- call handle_err(err,message)
-
- ! initialize canopy drip
- ! NOTE: canopy drip from the previous time step is used to compute throughfall for the current time step
- fluxStruct%hru(iHRU)%var(iLookFLUX%scalarCanopyLiqDrainage)%dat(1) = 0._dp 
-
- ! define the file if the first HRU
- if(iHRU==1) then
-  write(fileout,'(a,i0,a,i0,a)') trim(OUTPUT_PATH)//trim(OUTPUT_PREFIX)//'_spinup'//trim(output_fileSuffix)//'.nc'
-  call def_output(nHRU,nSoil(iHRU),fileout,err,message); call handle_err(err,message)
- endif
-
- ! write local model attributes and parameters to the model output file
- call writeAttrb(fileout,iHRU,attrStruct%hru(iHRU)%var,typeStruct%hru(iHRU)%var,err,message); call handle_err(err,message)
- call writeParam(fileout,iHRU,mparStruct%hru(iHRU)%var,bparStruct%var,err,message); call handle_err(err,message)
-
-end do  ! (looping through HRUs)
-
-! allocate space for the upslope area
-allocate(upArea(nHRU),stat=err); call handle_err(err,'problem allocating space for upArea')
+ ! loop through local HRUs
+ do iHRU=1,gru_struc(iGRU)%hruCount
+
+  ! check that the parameters are consistent
+  call paramCheck(mparStruct%gru(iGRU)%hru(iHRU)%var,err,message); call handle_err(err,message)
+  
+  ! calculate a look-up table for the temperature-enthalpy conversion 
+  call E2T_lookup(mparStruct%gru(iGRU)%hru(iHRU)%var,err,message); call handle_err(err,message)
+ 
+  ! read description of model initial conditions -- also initializes model structure components
+  ! NOTE: at this stage the same initial conditions are used for all HRUs -- need to modify
+  call read_icond(gru_struc(iGRU)%hruInfo(iHRU)%nSnow, & ! number of snow layers
+                  gru_struc(iGRU)%hruInfo(iHRU)%nSoil, & ! number of soil layers
+                  mparStruct%gru(iGRU)%hru(iHRU)%var,  & ! vector of model parameters
+                  indxStruct%gru(iGRU)%hru(iHRU),      & ! data structure of model indices
+                  progStruct%gru(iGRU)%hru(iHRU),      & ! model prognostic (state) variables
+                  err,message)                           ! error control
+  call handle_err(err,message)
+  
+  ! re-calculate height of each layer
+  call calcHeight(&
```

Comment:
Since `calcHeight` is rather generic, I would recommend renaming this to be more specific, e.e `calcLayerHeight()` or something like it.